### PR TITLE
[Kiali_269] Throw warning message when istio version is not supported

### PR DESCRIPTION
**Pull request by aljesusg**
[KIALI-269](https://issues.jboss.org/browse/KIALI-269)

The object will be warning_messages

## With a correct version
correct_version
## With a version not supported
version_error

### CODE REVIEW

File: status/status.go [hunk 1]
```diff
@@ -11,8 +11,9 @@ const (
 )
 
 type StatusInfo struct {
-	Status   map[string]string `json:"status"`
-	Products []ProductInfo     `json:"products"`
+	Status          map[string]string `json:"status"`
+	Products        []ProductInfo     `json:"products"`
+	WarningMessages []string          `json:"warning_messages"`
```
Comment (jotak) [inline status/status.go]: Please use camel-case in json: `warningMessages`.
All json has been changed now to conform to this rule (see https://issues.jboss.org/browse/KIALI-768 )

Comment (aljesusg) [reply status/status.go]: Ok, I didn't know that this apply to Json too

**Comment by jotak** [comment]
Can you also add unit tests please?

**Comment by abonas** [comment]
the title here says error message, but then in the code it is warning message, so which one it is? and how would that influence the users?

**Comment by aljesusg** [comment]
@abonas I think that we'll have a warning message in the message center to tell user that the Istio version installed is not supported by Kiali

**Comment by pilhuhn** [comment]
I think we should try to differentiate between "degraded" and "not working".
E.g. we require Istio 0.7.1 so it would not work with Istio 0.5.
But on the other hand it could work with Istio 0.8, but is degraded.
And yes, I am making those version numbers up.

**Comment by aljesusg** [comment]
@jotak tests added. Thanks

File: status/versions_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+	result = compareVersion("1.7.1", "0.8.2", ">")
+
+	if result {
+		t.Errorf("compareVersion was incorrect, got true, want false, 0.8.2 is not > 1.7.1")
+	}
+}
+
+func TestValidateVersion(t *testing.T) {
+	result := validateVersion(">= 0.7.1", "0.7.1")
```
Comment (jotak) [inline status/versions_test.go]: What if a version is "0.8.0-alpha" ? Or "0.8.0.1" ? Will it crash?

Comment (aljesusg) [reply status/versions_test.go]: Not with 0.8.0.1 but yes with 0.8.0-alpha. Will we use alpha versions?

Comment (jotak) [reply status/versions_test.go]: The user can install whatever he wants, no? I'm looking at our versions.go file: iiuc it's looking for pattern "[0-9]\\.[0-9]\\.[0-9]" (note that anything may follow this pattern - I'm not sure if the `FindStringSubmatch` used there truncates the version for the matched expression or takes the whole string like "0.8.0-alpha").
Also, looking at versions.go, version may be "Unknown". In which case I guess it should be considered as not supported.

**Comment by xeviknal** [comment]
Is it too overkilling use a library like that? Would it be useful?
https://github.com/hashicorp/go-version
It looks like it helps operating with version numbers.

**Comment by aljesusg** [comment]
I saw the lib too but for only check this I thought it  was not necessary.
@jotak I don't think that we should let users install alpha versions

**Comment by jotak** [comment]
But my point is we may not have any control on what is installed

**Comment by aljesusg** [comment]
You are right...ok so..I am gonna install the lib that @xeviknal said because that lib works with this kinds of versions too. Thanks @xeviknal and @jotak

**Comment by aljesusg** [comment]
Ok changed and added tests with alpha and longer releases examples.

**Comment by aljesusg** [comment]
Thanks @jotak

**Pull request merged**